src/libstd/ffi/c_str.rs

```diff
@@ -239,7 +239,7 @@ impl fmt::Display for NulError {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl From<NulError> for io::Error {
     fn from(_: NulError) -> io::Error {
-        io::Error::new(io::ErrorKind::InvalidInput,
+        io::Error::new(io::ErrorKind::InvalidData,
```

Comment:
This should probably stay as-is. `CString::new` returns a `NulError` when its argument (input) contains a null.